gix-diff/src/blob/mod.rs

```diff
@@ -5,6 +5,45 @@ use std::{collections::HashMap, path::PathBuf};
 use bstr::BString;
 pub use imara_diff::*;
 
+/// Re-export imara-diff v0.2 types for use with slider heuristics.
+///
+/// This module provides access to the v0.2 API of imara-diff, which includes
+/// support for Git's slider heuristics to produce more intuitive diffs.
+pub mod v0_2 {
+    pub use imara_diff_v0_2::*;
+}
+
+/// Compute a diff with Git's slider heuristics to produce more intuitive diffs.
+///
+/// This function uses `imara-diff` v0.2 which provides the [`v0_2::Diff`] structure
+/// that supports postprocessing with slider heuristics. The slider heuristics move
+/// diff hunks to more intuitive locations based on indentation and other factors,
+/// resulting in diffs that are more readable and match Git's output more closely.
+///
+/// # Examples
+///
+/// ```
+/// use gix_diff::blob::{diff_with_slider_heuristics, v0_2::{Algorithm, InternedInput}};
+///
+/// let before = "fn foo() {\n    let x = 1;\n}\n";
+/// let after = "fn foo() {\n    let x = 2;\n}\n";
+///
+/// let input = InternedInput::new(before, after);
+/// let diff = diff_with_slider_heuristics(Algorithm::Histogram, &input);
+///
+/// // The diff now has slider heuristics applied
+/// assert_eq!(diff.count_removals(), 1);
+/// assert_eq!(diff.count_additions(), 1);
```

Comment:
@Byron Counting removals and additions probably is not the best example to showcase slider heuristics as line numbers are not affected by moving sliders up and down. :-)